---
float32 xgoal
float32 ygoal
int32 goal
